api/tests/unit/environments/dynamodb/wrappers/test_unit_dynamodb_environment_v2_wrapper.py

```diff
@@ -53,6 +59,45 @@ def test_environment_v2_wrapper__get_identity_overrides_by_environment_id__retur
     assert results[0] == override_document
 
 
+def test_environment_v2_wrapper__get_identity_overrides_by_environment_id__last_evaluated_key__call_expected(
+    flagsmith_environments_v2_table: Table,
+    mocker: MockerFixture,
+) -> None:
+    # Given
+    wrapper = DynamoEnvironmentV2Wrapper()
+    mocker.patch.object(
+        wrapper, "get_table"
+    ).return_value = table_mock = mocker.MagicMock(spec=flagsmith_environments_v2_table)
+
+    last_evaluated_key = "next_page_key"
+    override_document = {"test": "document"}
+
+    table_mock.query.side_effect = [
+        {"Items": [override_document], "LastEvaluatedKey": last_evaluated_key},
+        {"Items": [override_document], "LastEvaluatedKey": None},
+    ]
+
+    # When
+    results = wrapper.get_identity_overrides_by_environment_id(
+        environment_id=mocker.ANY,
+        feature_id=mocker.ANY,
+    )
+
+    # Then
+    assert results == [override_document, override_document]
+    wrapper.table.query.assert_has_calls(
+        [
+            mocker.call(
+                KeyConditionExpression=mocker.ANY,
+            ),
+            mocker.call(
+                KeyConditionExpression=mocker.ANY,
+                ExclusiveStartKey=last_evaluated_key,
+            ),
+        ]
+    )
```

Comment:
So just to make sure I'm understanding this correctly: All the pages are returned in a single call. The response to the client could potentially span multiple documents that could be quite long?